internal/repo/datasource/sql_test.go

```diff
@@ -0,0 +1,357 @@
+					return
+				}
+				if !reflect.DeepEqual(rows[0].TransferTo, temp.TransferTo) {
+					t.Errorf("Want: %v, Got: %v", temp.TransferTo, rows[0].TransferTo)
+					return
+				}
+				if !reflect.DeepEqual(rows[0].Type, temp.Type) {
+					t.Errorf("Want: %v, Got: %v", temp.Type, rows[0].Type)
+					return
+				}
+				if !reflect.DeepEqual(rows[0].Status, temp.Status) {
+					t.Errorf("Want: %v, Got: %v", temp.Status, rows[0].Status)
+					return
+				}
+				if !reflect.DeepEqual(rows[0].Comment, temp.Comment) {
+					t.Errorf("Want: %v, Got: %v", temp.Comment, rows[0].Comment)
+					return
+				}
+
+			},
+		},
+		{
+			name: "failure::Get::scan error", //scan should return an error
+			filter: map[string]interface{}{
+				"user_id": "1234",
+			},
+			setupFunc: func() sqlDs {
+				db, mock, err := sqlmock.New()
+				if err != nil {
+					t.Fail()
+				}
+				dB := sqlDs{
+					sqlSvc: db,
+					table:  "newTemp",
+				}
+				mock.ExpectQuery(regexp.QuoteMeta("SELECT COUNT(`transaction_id`) FROM newTemp WHERE user_id = '1234'")).WillReturnError(nil).WillReturnRows(sqlmock.NewRows([]string{"transaction_id"}).AddRow("1").AddRow("2").AddRow("3"))
+				mock.ExpectQuery("SELECT transaction_id, account_number, user_id, amount, transfer_to, created_at, updated_at, status, type, comment FROM newTemp WHERE user_id = '1234' ORDER BY created_at LIMIT 1 OFFSET 2 ;").WillReturnRows(sqlmock.NewRows([]string{"transaction_id", "account_number", "user_id", "amount", "transfer_to", "created_at", "updated_at", "status", "type", "comment"}).AddRow(true, 1, "123", 1000, 1234567890, time.Now(), "abc", "approved", "debit", "no comments"))
+				return dB
+			},
+			validator: func(rows []model.Transaction, count int, err error) {
+				if !strings.Contains(err.Error(), "sql: Scan error on column") {
+					t.Errorf("Want: %v, Got: %v", "sql: Scan error on column", err.Error())
+				}
+			},
+		},
+		{
+			name:   "FAILURE:: query error",
+			filter: map[string]interface{}{"userid": "1234"},
+			setupFunc: func() sqlDs {
+				db, mock, err := sqlmock.New()
+				if err != nil {
+					t.Fail()
+				}
+				dB := sqlDs{
+					sqlSvc: db,
+					table:  "newTemp",
+				}
+				mock.ExpectQuery(regexp.QuoteMeta("SELECT COUNT(`transaction_id`) FROM newTemp WHERE userid = '1234'")).WillReturnError(nil).WillReturnRows(sqlmock.NewRows([]string{"transaction_id"}).AddRow("1").AddRow("2").AddRow("3"))
+				mock.ExpectQuery("SELECT transaction_id, account_number, user_id, amount, transfer_to, created_at, updated_at, status, type, comment FROM newTemp WHERE userid = '1234' ORDER BY created_at LIMIT 1 OFFSET 2 ;").WillReturnError(errors.New("Unknown column"))
+				return dB
+			},
+			validator: func(rows []model.Transaction, count int, err error) {
+				if !strings.Contains(err.Error(), "Unknown column") {
+					t.Errorf("Want: %v, Got: %v", "Unknown column", err)
+				}
+			},
+		},
+		{
+			name:   "FAILURE:: query error:: 2",
```

Comment:
What is this test case for? Descriptive names required.

Reply:
done